modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1022,7 +1023,21 @@ public final HitInfo hitTest(Point2D point) {
         TextLayout layout = getTextLayout();
         double x = point.getX() - getX();
         double y = point.getY() - getY() + getYRendering();
-        TextLayout.Hit h = layout.getHitInfo((float)x, (float)y, getText());
+        GlyphList[] runs = getRuns();
+        int runIndex = 0;
+        if (runs.length != 0) {
+            Point2D ptInParent = localToParent(x, y);
+            while (runIndex < runs.length - 1) {
+                if (ptInParent.getY() > runs[runIndex].getLocation().y
+                        && ptInParent.getY() < runs[runIndex + 1].getLocation().y) {
+                    break;
+                }
+                runIndex++;
+            }
+        }
+        int textRunStart = runs.length == 0 ? 0 : ((TextRun) runs[0]).getStart();
+        int curRunStart = runs.length == 0 ? 0 : ((TextRun) runs[runIndex]).getStart();
```

Comment:
very minor: we are computing run.length == 0 twice.  I know it might look neater than
```
int textRunStart;
int curRunStart;
if(runs.length == 0) {
  textRunStart = 0;
  curRunStart = 0;
} else {
 ...
}
```